Code/GraphMol/RGroupDecomposition/RGroupGa.cpp

```diff
@@ -0,0 +1,263 @@
+  child->mutate();
+  child->setOperationName(RgroupMutate);
+  child->decode();
+
+  auto& labelsToVarianceData = child->getLabelsToVarianceData();
+  if (labelsToVarianceData.size() == 0) return;
+#ifdef DEBUG
+  std::cerr << "RGroup mutate start" << std::endl;
+#endif
+#ifdef DEBUG
+  std::cerr << "Starting child score" << std::endl;
+  fingerprintVarianceScore(labelsToVarianceData);
+#endif
+
+  auto &parentPermutation = parent->getPermutation();
+  auto &childPermutation = child->getPermutation();
+  const auto& rgroupData = parent->getRGroupGA().getRGroupData();
+  const auto& matches = rgroupData.matches;
+  const auto& labels = rgroupData.labels;
+
+  for (auto pos = 0U; pos < parentPermutation.size(); pos++) {
+    int parentValue = parentPermutation.at(pos);
+    int childValue = childPermutation.at(pos);
+    if (parentValue != childValue) {
+      removeVarianceData(pos, parentValue, matches, labels,
+                         labelsToVarianceData);
+#ifdef DEBUG
+      std::cerr << "After removing parent" << std::endl;
+      fingerprintVarianceScore(labelsToVarianceData);
+#endif
+      addVarianceData(pos, childValue, matches, labels, labelsToVarianceData);
+#ifdef DEBUG
+      std::cerr << "After adding child" << std::endl;
+      fingerprintVarianceScore(labelsToVarianceData);
+#endif
+    }
+  }
+#ifdef DEBUG
+  std::cerr << "Final recalculating" << std::endl;
+  child->recalculateScore();
+  std::cerr << "RGroup mutate done" << std::endl;
+#endif
+}
+
+void RGroupGa::rGroupCrossoverOperation(
+    const std::vector<std::shared_ptr<RGroupDecompositionChromosome>>& parents,
+    std::vector<std::shared_ptr<RGroupDecompositionChromosome>>& children) {
+  auto parent1 = parents[0];
+  auto child1 = children[0];
+  auto parent2 = parents[1];
+  auto child2 = children[1];
+
+  child1->setOperationName(Crossover);
+  child2->setOperationName(Crossover);
+  clearVarianceData(child1->getLabelsToVarianceData());
+  clearVarianceData(child2->getLabelsToVarianceData());
+
+  parent1->twoPointCrossover(*parent2, *child1, *child2);
+}
+
+void RGroupGa::createOperations() {
+  // bias to mutation as that operator is so efficient
+  auto mutationOperation =
+      make_shared<GaOperation<RGroupDecompositionChromosome>>(
+          1, 1, 75.0, &rGroupMutateOperation);
+  auto crossoverOperation =
+      make_shared<GaOperation<RGroupDecompositionChromosome>>(
+          2, 2, 25.0, &rGroupCrossoverOperation);
+  operations.reserve(2);
+  operations.push_back(mutationOperation);
+  operations.push_back(crossoverOperation);
+}
+
+std::string timeInfo(const chrono::system_clock::time_point start) {
+  auto now = chrono::high_resolution_clock::now();
+  chrono::duration<double, milli> milliseconds = now - start;
+  auto seconds = milliseconds.count() / 1000.0;
+  auto format = boost::format("Time %7.2f") % seconds;
+  return format.str();
+}
+
+vector<vector<size_t>> RGroupGa::run() {
+  auto startTime = std::chrono::high_resolution_clock::now();

```

Comment:
high_resolution_clock::now returns a 
```
std::chrono::time_point<std::chrono::high_resolution_clock>
```
not a std::chrono:time_point, I wonder if this is causing build failures on win/osx?

Reply:
I'll look into moving to std::clock(), which seems a lot simpler.

Reply:
Moving to `std::clock` has fixed this issue.  All builds are successful now.